drivers/adc/ad9083/ad9083.c

```diff
@@ -0,0 +1,385 @@
+#include "delay.h"
+#include "clk.h"
+
+extern uint64_t clk_hz[][3];
+extern uint32_t vmax[];
+extern uint32_t fc[];
+extern uint8_t  rterm[];
+extern uint32_t en_hp[];
+extern uint32_t backoff[];
+extern uint32_t finmax[];
+extern uint64_t nco_freq_hz[][3];
+extern uint8_t  decimation[][4];
+extern uint8_t  nco0_datapath_mode[];
+extern adi_cms_jesd_param_t jtx_param[];
+extern int32_t adi_ad9083_jtx_startup(adi_ad9083_device_t *device,
+				      adi_cms_jesd_param_t *jtx_param);
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define CHIPID_AD9083		0x00EA
+#define CHIPID_MASK		0xFFFF
+#define SPI_IN_OUT_BUFF_SZ	0x3
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Read register.
+ * @param device - ad9083 device.
+ * @param reg - Address of the register.
+ * @param readval - Value of the register.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad9083_reg_get(struct ad9083_phy *device, uint32_t reg,
+		       uint8_t *readval)
+{
+	int32_t ret = 0;
+	uint8_t data[SPI_IN_OUT_BUFF_SZ] = {0};
+
+	if (reg < 0x1000) {
+		data[0] = (((reg >> 8) | 0x80) & 0xFF);
+		data[1] = (reg & 0xFF);
+		ret = spi_write_and_read(device->spi_desc, data, SPI_IN_OUT_BUFF_SZ);
+		if (ret != 0)
+			return ret;
+
+		*readval = data[2];
+	}
+	return SUCCESS;
+}
+
+/**
+ * @brief Write register.
+ * @param device - ad9083 device.
+ * @param reg - Address of the register.
+ * @param writeval - Value of the register.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t ad9083_reg_set(struct ad9083_phy *device, uint32_t reg,
+		       uint8_t writeval)
+{
+	int32_t ret = 0;
+	uint8_t data[SPI_IN_OUT_BUFF_SZ] = {0};
+
+	if (reg < 0x1000) {
+		data[0] = ((reg >> 8) & 0xFF);
+		data[1] = (reg & 0xFF);
+		data[2] = writeval;
+		ret = spi_write_and_read(device->spi_desc, data, SPI_IN_OUT_BUFF_SZ);
+		if (ret != 0)
+			return ret;
+	}
+	return SUCCESS;
+}
+
+/**
+ * Spi write and read compatible with ad9083 API
+ * @param user_data
+ * @param in_data - Pointer to array with the data to be sent on the SPI
+ * @param out_data - Pointer to array where the data to which the SPI will be written
+ * @param size_bytes - The size in bytes allocated for each of the indata and outdata arrays.
+ * @return 0 for success, any non-zero value indicates an error
+ */
+static int32_t ad9083_spi_xfer(void *user_data, uint8_t *in_data,
+			       uint8_t *out_data, uint32_t size_bytes)
+{
+	struct ad9083_phy *phy = user_data;
+	uint8_t data[6];
+	uint16_t bytes_number;
+	int32_t ret;
+	int32_t i;
+
+	bytes_number = (size_bytes & 0xFF);
```

Comment:
the type for `byte_number` is `uint16_t` but here you are applying only `0xFF` mask. Is this required?